### DISCUSSION TITLE
The output file from the ‘exportShmTreesWithNodes’ command appears to be missing some columns

### DISCUSSION THREAD

**Discussion by zongxiuXu** [milaboratory/mixcr#1224] [Q&A] [answered]
I did it! I am thrilled to share that I have successfully transitioned from my Master's to a Ph.D. program, having passed the necessary assessments. That's why you haven't heard from me for so long. Now, I'm back to continuing my SHM analysis.

In accordance with the tutorial available on your webpage (https://mixcr.com/mixcr/guides/b-cell-lineages-webinar/#visualizations), I diligently followed each step. Everything proceeded smoothly, apart from a minor issue where the  `Define function for drawing phylogenetic trees with amino acid sequence alignments` section lacked a `dev.off()` R code for the `pdf()` function.

However, I have reached an impasse with a particular step due to missing column names in the `lng_db` dataset. Upon investigating the origin of the issue, I identified that the `*_trees.txt` file, generated by the `exportShmTreesWithNodes` command, lacks the appropriate column names. The output file currently appears as follows:
[SHMtress_tsv_example.xlsx](https://github.com/milaboratory/mixcr/files/11933649/SHMtress_tsv_example.xlsx)

Having carefully reviewed the detailed explanation of the `exportShmTreesWithNodes` command on your website, I understood that the output file should include the column names I need, such as: `positionInReferenceOfFR2Begin`, `nMutationsCDR1BasedOnGermline`, and so on. Yet, despite my multiple attempts at executing the `exportShmTreesWithNodes` command using both `mixcr exportShmTreesWithNodes -f` and `mixcr exportShmTreesWithNodes --preset full`, the output file remains unchanged.

I humbly request your assistance in this matter! I feel tantalizingly close to successfully completing the entire SHM analysis process. Your expert guidance would be invaluable and greatly appreciated!!!

Below, you'll find the entirety of my code. Please feel free to disregard the internal naming conventions and loops for the time being.
[MiXCR4_fullLength_SHM.txt](https://github.com/milaboratory/mixcr/files/11933761/MiXCR4_fullLength_SHM.txt)
I'm working with bulk data, not single-cell sequencing. If the issue arises from this distinction, could you please guide me on how to obtain ‘nMutation’ and other relevant data?

**Comment by gnefedev** [comment, marked as answer]
Our congratulations!

For your case you need to use
`mixcr exportShmTreesWithNodes -p full -allPositionsInReference -allNMutations germline input.shmt`

**Reply by zongxiuXu** [reply]
Thank you very much for your assistance. I successfully executed the entire SHM analysis process following the method you provided. However, there were certain sections of the code that did not run successfully, which I adjusted based on my understanding, including `nMutations{CDR1Begin:FR3End}BasedOnGermline`. If this is a custom function, some parts may have been omitted. If it's not, kindly inform me if I've missed anything, if you have the time.

Moreover, I believe there might be issues with the mutation count from `CDR1` to `FR3`, as well as the regional assessment of these segments. For instance, if both `nMutationsCDR1BasedOnGermline` and `nMutationsFR3BasedOnGermline` have a mutation at the second position, according to the provided R code, they would be categorized as CDR1, resulting in the mutation at the second position being counted twice. I'm unsure if this is an error in my understanding.

Once more, I want to express my gratitude for your thorough response!

**Reply by gnefedev** [reply]
Hello!

`-allNMutations germline` and `-nMutations {CDR1Begin:FR3End} germline` will export mutations relative the same feature, i.e. from the start of this feature. I think, you expected mutations relative to `VRegion`. For this you should write it explicitly like this: `-nMutationsRelative FR3 VRegion`.

If you expected something else, or it's somehow not resolving your problem, please post your exact command and a sample of output that you think is not correct.

**Reply by zongxiuXu** [reply]
Oh, I see. I had thought that `nMutations{CDR1Begin:FR3End}BasedOnGermline` referred to multiple columns. In reality, it's a standalone column that can be generated by adjusting the parameter settings. I misunderstood. Thank you for your patience！
